extensions/amp-analytics/0.1/transport.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {assertHttpsUrl} from '../../../src/url';
+import {log} from '../../../src/log';
+import {loadPromise} from '../../../src/event-helper';
+
+/** @const {string} */
+const TAG_ = 'AmpAnalytics.Transport';
+
+/**
+ * @param {!Window} win
+ * @param {string} request
+ * @param {!Object<string, string>} transportOptions
+ */
+export function sendRequest(win, request, transportOptions) {
+  assertHttpsUrl(request);
+  if (transportOptions['beacon'] &&
+      Transport.sendRequestUsingBeacon(win, request)) {
+    return;
+  }
+  if (transportOptions['xhrpost'] &&
+      Transport.sendRequestUsingXhr(win, request)) {
+    return;
+  }
+  if (transportOptions['image']) {
+    Transport.sendRequestUsingImage(win, request);
+    return;
+  }
+  log.warn(TAG_, 'Failed to send request', request, transportOptions);
+}
+
+/**
+ * @visibleForTesting
+ */
+export class Transport {
+
+  /**
+   * @param {!Window} win
+   * @param {string} request
+   */
+  static sendRequestUsingImage(win, request) {
+    const image = new Image();
+    image.src = request;
```

Comment:
the URL would be up to the publisher configs, best not to make much assumptions but give flexibility then provide assumptions in default configs in `vendors.js`